samples/subsys/usb/hid/src/main.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2016 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_LEVEL 4
+#define SYS_LOG_DOMAIN "main"
+#include <logging/sys_log.h>
+
+#include <zephyr.h>
+
+#include <usb/usb_device.h>
+#include <usb/class/usb_hid.h>
+
+/* Some HID sample Report Descriptor */
+static const u8_t hid_report_desc[] = {
+	0x05, 0x01,		/* USAGE_PAGE (Generic Desktop)		*/
+	0x09, 0x00,		/* USAGE (Undefined)			*/
+	0xa1, 0x01,		/* COLLECTION (Application)		*/
```

Comment:
Please add the defines|macros for Global Items, Collection...  to usb_hid.h and at least the defines for values you have used (where it makes sense).

Reply:
I would prefer not to touch this, since this is how it is represented in all samples, if I have simple defines probably they cannot be reused in other code.

Reply:
'''
#define USAGE_GENERIC_DESKTOP	0x01

#define USAGE_PAGE(type)	0x05, type
'''
like this?

Reply:
something like that:
```

#define HID_ITEM(bTag, bType, bSize)        ((bTag & 0xf) << 4) | (bType & 0x3) << 2) | (size & 0x3))
#define HID_GLOBAL_ITEM(bTag, bSize)        HID_ITEM(...)
#define HID_GI_PAGE                         HID_GLOBAL_ITEM(...)

```
The macros for the Items would increase readability enormously.

Reply:
How are you going handle long items here?

Reply:
More precisely, the macros above are for the prefix. Maybe it would be better to add PRFX to the name? e.g. HID_ITEM_PRFX

Reply:
```
/* HID Items types */
#define ITEM_MAIN	0x0
#define ITEM_GLOBAL	0x1
#define ITEM_LOCAL	0x2

/* HID Global Items tags */
#define ITEM_TAG_USAGE_PAGE	0x0

/* HID Local Items tags */
#define ITEM_TAG_USAGE		0x0

#define HID_ITEM(bTag, bType, bSize)	(((bTag & 0xf) << 4) | \
					 ((bType & 0x3) << 2) | (bSize & 0x3))

#define HID_GLOBAL_ITEM(bTag, bSize)	HID_ITEM(bTag, ITEM_GLOBAL, bSize)
#define HID_LOCAL_ITEM(bTag, bSize)	HID_ITEM(bTag, ITEM_LOCAL, bSize)

#define HID_GI_USAGE_PAGE		HID_GLOBAL_ITEM(ITEM_TAG_USAGE_PAGE, \
							0x01)
#define HID_LI_USAGE			HID_LOCAL_ITEM(ITEM_TAG_USAGE, 0x01)

/* Defined in Universal Serial Bus HID Usage Tables version 1.11 */
#define USAGE_GEN_DESKTOP		0x01

/* Generic Desktop Page usages */
#define USAGE_GEN_DESKTOP_UNDEFINED	0x00
```

And then in the report descriptor

```
static const u8_t hid_report_desc[] = {
	/* 0x05, 0x01, 		USAGE_PAGE (Generic Desktop)		*/
	HID_GI_USAGE_PAGE, USAGE_GEN_DESKTOP,
	/* 0x09, 0x00,		USAGE (Undefined)			*/
	HID_LI_USAGE, USAGE_GEN_DESKTOP_UNDEFINED,
```

Reply:
looks good, thx

Reply:
OK, I will change to this system